runtime/src/prioritization_fee_cache.rs

```diff
@@ -0,0 +1,788 @@
+                entry
+            }
+        }
+    }
+
+    /// Update with a list of transactions' tx_priority_details and tx_account_locks; Only
+    /// transactions have both valid priority_detail and account_locks will be used to update
+    /// fee_cache asynchronously.
+    pub fn update<'a>(&self, bank: Arc<Bank>, txs: impl Iterator<Item = &'a SanitizedTransaction>) {
+        let mut successful_transaction_update_count: u64 = 0;
+        let (_, send_updates_time) = measure!(
+            {
+                for sanitized_transaction in txs {
+                    let priority_details = sanitized_transaction.get_transaction_priority_details();
+                    let account_locks = sanitized_transaction
+                        .get_account_locks(bank.get_transaction_account_lock_limit());
+
+                    if priority_details.is_none() || account_locks.is_err() {
+                        continue;
+                    }
+
+                    let writable_accounts = Arc::new(
+                        account_locks
+                            .unwrap()
+                            .writable
+                            .iter()
+                            .map(|key| **key)
+                            .collect::<Vec<_>>(),
+                    );
+
+                    self.sender
+                        .send(FinalizingServiceUpdate::TransactionUpdate {
+                            slot: bank.slot(),
+                            transaction_fee: priority_details.unwrap().priority,
+                            writable_accounts,
+                        })
+                        .unwrap_or_else(|err| {
+                            warn!(
+                                "prioritization fee cache transaction updates failed: {:?}",
+                                err
+                            );
+                        });
+                    saturating_add_assign!(successful_transaction_update_count, 1)
+                }
+            },
+            "send_updates",
+        );
+
+        self.metrics
+            .accumulate_total_update_elapsed_us(send_updates_time.as_us());
+        self.metrics
+            .accumulate_successful_transaction_update_count(successful_transaction_update_count);
+    }
+
+    /// Finalize prioritization fee when it's bank is completely replayed from blockstore,
+    /// by pruning irrelevant accounts to save space, and marking its availability for queries.
+    pub fn finalize_priority_fee(&self, slot: Slot) {
+        self.sender
+            .send(FinalizingServiceUpdate::BankFrozen { slot })
+            .unwrap_or_else(|err| {
+                warn!(
+                    "prioritization fee cache signalling bank frozen failed: {:?}",
+                    err
+                )
+            });
+    }
+
+    /// Internal function is invoked by worker thread to update slot's minimum prioritization fee,
+    /// Cache lock contends here.
+    fn update_transaction(
```

Comment:
This fn name makes less sense now. The thing being updated is the cache, and there isn't a transaction passed in.
Could it just be `update_cache`?

Reply:
✅